cmd/rook/filesystem-create.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2016 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package rook
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/rook/rook/pkg/model"
+	"github.com/rook/rook/pkg/rook/client"
+	"github.com/rook/rook/pkg/util/flags"
+	"github.com/spf13/cobra"
+)
+
+var (
+	newFilesystemName string
+	newFilesystemPool string
+)
+
+var filesystemCreateCmd = &cobra.Command{
+	Use:   "create",
+	Short: "Creates a new shared filesystem in the cluster",
+}
+
+func init() {
+	filesystemCreateCmd.Flags().StringVarP(&newFilesystemName, "name", "n", "", "Name of new filesystem to create (required)")
+	filesystemCreateCmd.Flags().StringVarP(&newFilesystemPool, "pool-name", "p", "", "Name of storage pool to create new filesystem in (defaults to name of filesystem)")
```

Comment:
i actually waffled on this one for a while because under the covers ceph actually allows multiple data pools (but only a single metadata pool).  we may want to eventually allow the user to specify which pool they want to use for metadata and which pools (plural) they want to use for data, instead of creating new pools during orchestration.

for now, we can probably just remove the pool-name option and generate 1 metadata and 1 data pool based off the file system name.